lib/http/client.cpp

```diff
@@ -962,6 +964,56 @@ Client::send_to_device(const std::string &event_type,
         put<nlohmann::json>(api_path, body, callback);
 }
 
+void
+Client::get_room_visibility(const std::string &room_id,
+                            Callback<mtx::responses::RoomVisibility> cb)
+{
+        const auto api_path = "/client/r0/directory/list/room/" +
+                               mtx::client::utils::url_encode(room_id);
+
+        get<mtx::responses::RoomVisibility>(api_path,
+                                            [cb](const mtx::responses::RoomVisibility &res,
+                                                  HeaderFields,
+                                                  RequestErr err) { cb(res, err); });
+}
+
+void 
+Client::put_room_visibility(const std::string &room_id,
+                                 const nlohmann::json &body,
```

Comment:
This should not take a json.

Reply:
Why not? If I'm not wrong, [one of the interfaces for a PUT](https://github.com/Nheko-Reborn/mtxclient/blob/c5f6f422ab1559db330d3a711dff424d9bb200be/lib/http/client.cpp#L163) takes a JSON.

Reply:
Because all of our APIs take structs, so that you get a compiler error, when you make a mistake.